libraries/AP_RCProtocol/AP_RCProtocol_FPort.cpp

```diff
@@ -37,8 +38,9 @@ extern const AP_HAL::HAL& hal;
 #define FLAGS_FAILSAFE_BIT	3
 #define FLAGS_FRAMELOST_BIT	2
 
-#define CHAN_SCALE_FACTOR ((2000.0 - 1000.0) / (1800.0 - 200.0))
-#define CHAN_SCALE_OFFSET (int)(1000.0 - (CHAN_SCALE_FACTOR * 200.0 + 0.5f))
+#define CHAN_SCALE_FACTOR1 1000U
```

Comment:
You really could have put *real* types on these while you were at it...
```suggestion
static const uint16_t CHAN_SCALE_FACTOR1 = 1000;
```

Reply:
using defines or types really makes no difference, and other code in this library uses defines, and is a bit shorter (and imo clearer)